### PR TITLE
Add details of changes to push event

### PR DESCRIPTION
**Pull request by stephenc**
For the [Jenkins Bitbucket Branch Source Plugin](https://github.com/jenkinsci/bitbucket-branch-source-plugin) to respond to events in a performant manner, we need the details of the changes in the push request.

@reviewbybees

### CODE REVIEW

**Comment by stephenc** [comment]
Example event before change:

```
{
	"repository": {
		"scmId": "git",
		"project": {
			"key": "PROJECT_1",
			"name": "Project 1"
		},
		"slug": "rep_1",
		"public": false,
		"ownerName": "PROJECT_1",
		"fullName": "PROJECT_1/rep_1",
		"owner": {
			"username": "PROJECT_1",
			"displayName": "PROJECT_1"
		}
	}
}
```

Example event after change:

```
{
	"actor": {
		"username": "admin",
		"displayName": "Administrator"
	},
	"repository": {
		"scmId": "git",
		"project": {
			"key": "PROJECT_1",
			"name": "Project 1"
		},
		"slug": "rep_1",
		"public": false,
		"ownerName": "PROJECT_1",
		"fullName": "PROJECT_1/rep_1",
		"owner": {
			"username": "PROJECT_1",
			"displayName": "PROJECT_1"
		}
	},
	"push": {
		"changes": [{
			"created": false,
			"closed": true,
			"old": {
				"type": "branch",
				"name": "hotfix/abra-cadabra!/111",
				"target": {
					"type": "commit",
					"hash": "667326dfff4931f0f6d5019d7f1590c1fa4ac214"
				}
			},
			"new": null
		}]
	}
}
```

The event payload has been aligned as closely as I can get it with the [event payload for Bitbucket Cloud](https://confluence.atlassian.com/bitbucket/event-payloads-740262817.html#expander-control-887398559)

**Comment by rsandell** [comment]
:bee:

File: src/main/java/nl/topicus/bitbucket/events/BitbucketPushChange.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ */
+package nl.topicus.bitbucket.events;
+
+import com.atlassian.bitbucket.repository.RefType;
+import com.atlassian.bitbucket.repository.StandardRefType;
+import org.codehaus.jackson.map.annotate.JsonSerialize;
+
+public class BitbucketPushChange {
+    private Object _new;
+    private Object _old;
```
Comment (jglick) [inline src/main/java/nl/topicus/bitbucket/events/BitbucketPushChange.java]: Should these not be declared of type `State`? If there is a reason for `Object`, please leave a comment explaining why, unless there is already some higher-level documentation covering that.

File: src/main/java/nl/topicus/bitbucket/events/BitbucketPushChange.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        }
+
+        public void setTarget(Target target) {
+            this.target = target;
+        }
+
+        public void setType(RefType type) {
+            for (StandardRefType t : StandardRefType.values()) {
+                if (t.equals(type)) {
+                    this.type = t.name().toLowerCase();
```
Comment (jglick) [inline src/main/java/nl/topicus/bitbucket/events/BitbucketPushChange.java]: Use `Locale.ENGLISH` to be safe, though neither [`BRANCH` nor `TAG`](https://developer.atlassian.com/static/javadoc/bitbucket-server/4.8.1/api/reference/com/atlassian/bitbucket/repository/StandardRefType.html) contain the letter `I`.

**Comment by stephenc** [comment]
@jglick no downstream PR yet. I need confirmation that this plugin is accepting this event model change before I can upgrade our test server and validate the PR against it. The data model matches the required elements from the Bitbucket cloud event model, so I am confident that this will address the issue blocking my downstream PR

**Comment by stephenc** [comment]
@reviewbybees done

**Comment by amuniz** [comment]
LGTM :+1:

**Comment by stephenc** [comment]
@Eernie could you let us know your plans w.r.t. this PR?

**Comment by Eernie** [comment]
yes i'll try to look in to this, this weekend. For now: I don't see anything wrong with it:)

**Comment by stephenc** [comment]
@Eernie cool, I'll go ahead and write the matching changes in the Jenkins plugin so

**Comment by Eernie** [comment]
LGTM! Lets merge

**Comment by stephenc** [comment]
w00t

**Comment by casz** [comment]
@Eernie Could a release would be nice since https://groups.google.com/forum/#!topic/jenkinsci-dev/jyN1wzNX1T4

**Comment by Eernie** [comment]
@casz Done. Maybe a bit late, sorry for that.

**Comment by casz** [comment]
No need to be sorry! Was just a gentle or rather not so gentle reminder 🗡  

Awesome! 🚀

**Comment by burnettk** [comment]
this PR may have caused #31.

**Pull request merged**